spirv_cross.cpp

```diff
@@ -977,19 +1013,18 @@ const SPIRType &Compiler::get_type_from_variable(uint32_t id) const
 uint32_t Compiler::get_non_pointer_type_id(uint32_t type_id) const
 {
 	auto *p_type = &get<SPIRType>(type_id);
-	while (p_type->pointer)
+	if (p_type->pointer)
```

Comment:
I assume this is because you want double-pointers (OpVariable of pointer) to become just a plain pointer?

Reply:
Would it make sense to rename the method to be clearer about its intent?

Reply:
That is correct, and it probably would be. I just wanted to reduce the size of this change; it's already pretty big.